sdk/src/rplidar_driver.cpp

```diff
@@ -1792,10 +1792,10 @@ u_result RPlidarDriverImplCommon::grabScanData(rplidar_response_measurement_node
 
     switch (_dataEvt.wait(timeout))
     {
-    case rp::hal::Event::EVENT_TIMEOUT:
+    case static_cast<unsigned long>(rp::hal::Event::EVENT_TIMEOUT):
```

Comment:
enum
    {
        EVENT_OK = 1,
        EVENT_TIMEOUT = -1,
        EVENT_FAILED = 0,
    };
    It should not be unsigned

Reply:
Oh, sorry, i updated too fast... I though the default value was -1.

Reply:
They actually do the same for windows...
https://github.com/rapyuta-robotics/rplidar_ros/blob/rr-devel/sdk/src/hal/event.h#L108

Reply:
yup they should have used either an int8 or preferably use enum class. Not sure if this was already fixed upstream but can contribute the fix if it was not handled properly.